examples/solids/elasticity.c

```diff
@@ -115,8 +128,37 @@ int main(int argc, char **argv) {
 
   // Wrap context in libCEED objects
   CeedQFunctionContextCreate(ceed, &ctx_phys);
-  CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
-                              sizeof(*phys), phys);
+  switch (app_ctx->problem_choice) {
+  case ELAS_LINEAR:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
+  case ELAS_SS_NH:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
+  case ELAS_FSInitial_NH1:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
+  case ELAS_FSInitial_NH2:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
+  case ELAS_FSCurrent_NH1:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
+  case ELAS_FSCurrent_NH2:
+    CeedQFunctionContextSetData(ctx_phys, CEED_MEM_HOST, CEED_USE_POINTER,
+                                sizeof(*phys), phys);
+    break;
```

Comment:
Why not fallthrough to reduce repetition?

Reply:
Agreed for concision, but I also want this switch eliminated and attempted sharing has caused problems before (when one model is extended).

Reply:
Yeah, I think the original design is straining to keep up with the extensions we've made to the example.

Reply:
Certainly, but imposing long-term abstractions when starting the example could have been paralyzing. I think it's become overdue now though.